docs/Ramm.md

```diff
@@ -0,0 +1,186 @@
+
+# Ramm Contract Developer Documentation
+
+## Overview
+
+The `Ramm` contract is designed to manage liquidity and facilitate swaps between NXM tokens and ETH. It provides a dynamic pricing mechanism through ratcheting and circuit breakers to maintain stability. The contract includes features for managing ETH and NXM reserves, along with ensuring that swaps operate within predefined parameters and limits.
+
+## Key Concepts
+
+### Slot0 and Slot1
+`Slot0` and `Slot1` manage the contract's internal state for liquidity, ETH reserves, and NXM reserves. They store important values such as the available budget for liquidity injection and the current reserves.
+
+```solidity
+struct Slot0 {
+  uint128 nxmReserveA;
+  uint128 nxmReserveB;
+}
+
+  struct Slot1 {
+    uint128 ethReserve;
+    uint88 budget;
+    uint32 updatedAt;
+    bool swapPaused;
+  }
+```
+
+### Circuit Breakers
+Circuit breakers are in place to limit the total amount of ETH and NXM that can be released within a given period.
+
+| Parameter     | Description                                  |
+|---------------|----------------------------------------------|
+| `ethReleased` | Amount of ETH released                       |
+| `ethLimit`    | Maximum ETH that can be released before halt |
+| `nxmReleased` | Amount of NXM released                       |
+| `nxmLimit`    | Maximum NXM that can be released before halt |
+
+### Observations
+The `Observation` struct is used to track historical prices, allowing for Time-Weighted Average Price (TWAP) calculations.
+
+```solidity
+struct Observation {
+  uint32 timestamp;
+  uint112 priceCumulativeAbove;
+  uint112 priceCumulativeBelow;
+}
+```
+
+### Liquidity Management Parameters
+Several constants manage the liquidity behavior and price adjustments:
+
+```solidity
+uint public constant LIQ_SPEED_PERIOD = 1 days;
+uint public constant TARGET_LIQUIDITY = 5_000 ether;
+uint public constant FAST_LIQUIDITY_SPEED = 1_500 ether;
+uint public constant NORMAL_RATCHET_SPEED = 400;
+uint public constant FAST_RATCHET_SPEED = 5_000;
+```
+
+## Mutative Functions
+
+### `swap`
+Allows users to swap NXM tokens for ETH or vice versa.
```

Comment:
This function can and probably should be used from offchain to estimate the amount of nxm or eth one would if they swap by using `ramm.callStatic.swap(nxmIn, 0, future_timestamp)` (`eth_call`). They should add the slippage when swapping.